File: Semestre_4_2019_2020/INFO0402_C++/TP/algos/algo3/main.c
#include <stdio.h>

int main() {
    int taille, i=0, j=0;
    //Initialisation de taille
    printf("Entrez la taille de la matrice:\n");
    scanf("%d", &taille);
    //Initialisation des matrices 1 et 2
    int M1[taille][taille], M2[taille][taille], M3[taille][taille];
    printf("Entrez les valeurs de la matrice 1:\n");
    for(i=0; i<taille; i++){
        for(j=0; j<taille; j++){
            scanf("%d", &M1[i][j]);
        }
    }
    printf("Entrez les valeurs de la matrice 2:\n");
    for(i=0; i<taille; i++){
        for(j=0; j<taille; j++){
            scanf("%d", &M2[i][j]);
        }
    }
    //Initialisation de M3
    for(i=0; i<taille; i++){
        for(j=0; j<taille; j++){
            if(M1[i][j]==1 && M2[i][j]==1){
                M3[i][j]=1;
            }
            else{
                M3[i][j]=0;
            }
        }
    }
    //Affichage M3
    for(i=0; i<taille; i++){
        for(j=0; j<taille; j++){
            printf("%d ", M3[i][j]);
        }
        printf("\n");
    }
    return 0;
}

File: Semestre_4_2019_2020/INFO0402_C++/TP/algos/algo3/main.java
import java.util.Scanner;

class main{
    public static void main (String [] args) {
        Scanner clavier = new Scanner(System.in);

        //initialisation des variables usuelles
        int taille, i, j;

        //initialisation de taille
        System.out.println("entrez la taille de la matrice :");
        taille=clavier.nextInt();

        //initialisation des matrices M1 et M2
        int [][] M1= new int [taille][taille];
        int [][] M2= new int [taille][taille];
        int [][] M3= new int [taille][taille];
        System.out.println("entrez les valeurs de la matrice M1 :");
        for(i=0;i<taille;i++){
            for(j=0;j<taille;j++){
                System.out.print("M1["+i+"]["+j+"] = ");
                M1[i][j]=clavier.nextInt();
            }
        }
        System.out.println("entrez les valeurs de la matrice M2 :");
        for(i=0;i<taille;i++){
            for(j=0;j<taille;j++){
                System.out.print("M2["+i+"]["+j+"] = ");
                M2[i][j]=clavier.nextInt();
            }
        }

        //initialisation de M3
        for(i=0;i<taille;i++){
            for(j=0;j<taille;j++){
                if(M1[i][j]==1 && M2[i][j]==1){
                    M3[i][j]=1;
                }
                else{
                    M3[i][j]=0;
                }
            }
        }


        //affichage M3
        System.out.println("M3 :");
        for(i=0;i<taille;i++){
            for(j=0;j<taille;j++){
                System.out.print(" "+M3[i][j]);
            }
            System.out.println();
        }
    }
}